spring-boot/proxy/src/main/java/eu/coatrack/proxy/security/ApiKeyAuthTokenVerifier.java

```diff
@@ -20,166 +20,124 @@
-            }
-        } catch (HttpClientErrorException e) {
-            if (e.getStatusCode().equals(HttpStatus.NOT_FOUND)) {
-                log.debug("API key value is invalid: " + apiKeyValue);
-                return false;
-            } else {
-                log.error("Error when communicating with auth server", e);
-                throw new AuthenticationServiceException("Communication with auth server failed");
-            }
-        }
+    private Authentication createApiKeyAuthToken(Authentication authentication) {
+        String apiKeyValue = getApiKeyValue(authentication);
+        //TODO this is just a workaround for now: check for fixed API key to allow CoatRack admin access
+        if (isAdminsKey(apiKeyValue)) {
+            return createAdminsAuthToken(apiKeyValue);
+        } else
+            return verifyApiKeyAndIfAuthorizedCreateConsumerAuthToken(apiKeyValue);
     }
 
-    private ServiceApi getServiceApiByApiKey(String apiKeyValue) {
+    private String getApiKeyValue(Authentication authentication) {
+        log.debug("Getting API key value from authentication {}.", authentication.getName());
+        Assert.notNull(authentication.getCredentials());
+        Assert.isInstanceOf(String.class, authentication.getCredentials());
+        String apiKeyValue = (String) authentication.getCredentials();
+        Assert.hasText(apiKeyValue);
+        return apiKeyValue;
+    }
 
-        try {
-            log.debug("trying to get service api entity by APIKEY value {}", apiKeyValue);
+    private boolean isAdminsKey(String apiKeyValue) {
+        log.debug("Checking if the API with the value {} is admins API key.", apiKeyValue);
+        return apiKeyValue.equals(ApiKey.API_KEY_FOR_YGG_ADMIN_TO_ACCESS_PROXIES);
+    }
 
-            String urlToGetServiceApi = securityUtil.attachGatewayApiKeyToUrl(
-                    adminBaseUrl + adminResourceToGetServiceByApiKeyValue + apiKeyValue);
+    private Authentication createAdminsAuthToken(String apiKeyValue) {
+        log.debug("Creating admins authentication token using API key with the value {}.", apiKeyValue);
+        Set<SimpleGrantedAuthority> authoritiesGrantedToYggAdmin = new HashSet<>();
+        authoritiesGrantedToYggAdmin.add(new SimpleGrantedAuthority(
+                ServiceApiAccessRightsVoter.ACCESS_SERVICE_AUTHORITY_PREFIX + "refresh"));
+        ApiKeyAuthToken apiKeyAuthToken = new ApiKeyAuthToken(apiKeyValue, authoritiesGrantedToYggAdmin);
+        apiKeyAuthToken.setAuthenticated(true);
+        log.info("Admin is successfully authenticated using the API key with the value {}.", apiKeyValue);
+        return apiKeyAuthToken;
+    }
 
-            ResponseEntity<ServiceApi> responseEntity = restTemplate.getForEntity(urlToGetServiceApi, ServiceApi.class);
+    private Authentication verifyApiKeyAndIfAuthorizedCreateConsumerAuthToken(String apiKeyValue) {
```

Comment:
Maybe split this in two logical parts?
- first a method `boolean isApiKeyValid(String apiKeyValue)`
- then a method `Authentication createConsumerAuthToken(String apiKeyValue)`

Reply:
I renamed the function to "createConsumerAuthTokenIfApiKeyIsAuthorized". However, I found this name to be very descriptive and the the logic of validating an API key and creating a AuthToken is already separated to different functions.